packages/components/nodes/multimodal/OpenAI/VLLMChain.ts

```diff
@@ -0,0 +1,201 @@
+import { OpenAI as OpenAIClient, ClientOptions } from 'openai'
+import { BaseChain, ChainInputs } from 'langchain/chains'
+import { ChainValues } from 'langchain/schema'
+import { BasePromptTemplate, ChatPromptTemplate, SystemMessagePromptTemplate } from 'langchain/prompts'
+import path from 'path'
+import { getUserHome } from '../../../src/utils'
+import fs from 'fs'
+
+/**
+ * Interface for the input parameters of the OpenAIVisionChain class.
+ */
+export interface OpenAIVisionChainInput extends ChainInputs {
+    openAIApiKey?: string
+    openAIOrganization?: string
+    throwError?: boolean
+    prompt?: BasePromptTemplate
+    configuration?: ClientOptions
+    imageUrls?: []
+    imageResolution?: string
+    temperature?: number
+    modelName?: string
+    maxTokens?: number
+    topP?: number
+    whisperConfig?: any
+}
+
+/**
+ * Class representing a chain for generating text from an image using the OpenAI
+ * Vision API. It extends the BaseChain class and implements the
+ * OpenAIVisionChainInput interface.
+ */
+export class VLLMChain extends BaseChain implements OpenAIVisionChainInput {
+    static lc_name() {
+        return 'VLLMChain'
+    }
+    prompt: BasePromptTemplate | undefined
+
+    inputKey = 'input'
+    outputKey = 'text'
+    imageUrls?: []
+    imageResolution: string = 'low'
+    openAIApiKey?: string
+    openAIOrganization?: string
+    clientConfig: ClientOptions
+    client: OpenAIClient
+    throwError: boolean
+    temperature?: number
+    modelName?: string
+    maxTokens?: number
+    topP?: number
+
+    whisperConfig?: any
+
+    constructor(fields: OpenAIVisionChainInput) {
+        super(fields)
+        this.throwError = fields?.throwError ?? false
+        this.imageResolution = fields?.imageResolution ?? 'low'
+        this.openAIApiKey = fields?.openAIApiKey
+        this.prompt = fields?.prompt
+        this.temperature = fields?.temperature
+        this.modelName = fields?.modelName
+        this.maxTokens = fields?.maxTokens
+        this.topP = fields?.topP
+        this.imageUrls = fields?.imageUrls ?? []
+        this.whisperConfig = fields?.whisperConfig ?? {}
+        if (!this.openAIApiKey) {
+            throw new Error('OpenAI API key not found')
+        }
+
+        this.openAIOrganization = fields?.openAIOrganization
+
+        this.clientConfig = {
+            ...fields?.configuration,
+            apiKey: this.openAIApiKey,
+            organization: this.openAIOrganization
+        }
+
+        this.client = new OpenAIClient(this.clientConfig)
+    }
+
+    async _call(values: ChainValues): Promise<ChainValues> {
+        const userInput = values[this.inputKey]
+
+        const vRequest: any = {
```

Comment:
can we implement the interface, would love avoiding using `any` if possible

Reply:
Updated.